### DISCUSSION TITLE
Data pagamento in DDT

### DISCUSSION THREAD

**Discussion by FrantZ1971** [fattureincloud/api#418] [Q&A] [answered]
Buongiorno.
Invocando il metedo issued_documents per il tipo delivery_note non riesco a valorizzare la data di pagamento.
Ho provato con payments_list [] oppure con dn_date ma nella risposta il campo next_due_date risulta sempre NULL.
Grazie per qualsiasi suggerimento

**Comment by emajo** [comment]
Ciao @FrantZ1971, la data del pagamento va settata nel campo `data.payments_list[].due_date`, se il campo `next_due_date` risulta null può darsi sia dovuto al fatto che il pagamento sia stato settato a `paid`.

In caso di problemi le consiglio di condividerci il json della richiesta.

Grazie,
Emanuele.

**Comment by FrantZ1971** [comment]
Ciao, grazie per il riscontro. Forse sto sbagliando qualcosa perchè non la vedo nella risposta.
Questo il json di richiesta:

{
    "data": {
        "type": "delivery_note",
        "use_gross_prices": false,
        "number": 19944,
        "date": "2023-06-29",
        "entity": {
            "id": 19944,
            "name": "XXX",
            "address_street": "XXX",
            "address_postal_code": "XXX",
            "address_city": "XXX",
            "address_province": "XX",
            "code": "1539",
            "vat_number": "XXX"
        },
        "items_list": [
            {
                "code": "885",
                "name": "XXX",
                "qty": 60,
                "net_price": 12.5,
                "discount": 10.0
            }
        ],
        "payments_list": [
            {
                "due_date": "2024-06-01",
                "amount": 675,
                "status": "not_paid"
            }
        ]
    }
}

Nella response trovo 
        "next_due_date": null,

Grazie per il supporto

**Comment by emajo** [comment, marked as answer]
Ciao @FrantZ1971, si scusami per la svista ma il campo dei pagamenti non risulta settabile per il tipo di documento `delivery_note`, probabilmente il modo più corretto di fare quello ciò che dici è creare una fattura con allegato il ddt, come è possibile fare anche dalla nostra versione web.

**Reply by valmoz** [reply]
Buongiorno @FrantZ1971 possiamo chiudere la discussione?
Grazie
Mauro

**Comment by FrantZ1971** [comment]
Si, grazie


  Francesco Rattazzi

  Senior Manager

  Mobile: +39 335 69 78 477

[cid:f35e1fc1-5866-439a-8307-faf23d7de373]

www.biosmanagement.com<http://www.biosmanagement.com/>

Headquarter: Santa Vittoria d’Alba

Business Hubs: Barcelona|London|Lugano|Madrid|Milan|Rome|Turin

[cid:11f96575-d986-425a-be21-d66a09aeffa5][cid:d6ebdceb-5364-4d76-a4d9-2b0863d78efb]





________________________________
Da: Mauro Valota ***@***.***>
Inviato: venerdì 8 marzo 2024 14:15
A: fattureincloud/api ***@***.***>
Cc: Francesco Rattazzi - Bios ***@***.***>; Mention ***@***.***>
Oggetto: Re: [fattureincloud/api] Data pagamento in DDT (Discussion #418)


Buongiorno @FrantZ1971<https://github.com/FrantZ1971> possiamo chiudere la discussione?
Grazie
Mauro
